# Question 1

pipenv version: 2021.5.29 

# Question 2

sha256:121f78d6564000dc5e968394f45aac87981fcaaf2be40cfcd8f07b2baa1e1829

# Question 3 - Probability of churning (Script)

In [10]:
import pickle

In [11]:
input_file =  'model1.bin'
input_file_dv =  'dv.bin'

In [12]:
with open(input_file, 'rb') as f_in:
    model = pickle.load(f_in)
    
with open(input_file_dv, 'rb') as f_in:
    dv = pickle.load(f_in)

C:\Users\Rajit_Sanghvi\Anaconda3\lib\site-packages\sklearn\base.py:315: UserWarning: Trying to unpickle estimator LogisticRegression from version 1.0 when using version 0.24.2. This might lead to breaking code or invalid results. Use at your own risk.
  UserWarning)
C:\Users\Rajit_Sanghvi\Anaconda3\lib\site-packages\sklearn\base.py:315: UserWarning: Trying to unpickle estimator DictVectorizer from version 1.0 when using version 0.24.2. This might lead to breaking code or invalid results. Use at your own risk.
  UserWarning)


In [13]:
model, dv

(LogisticRegression(), DictVectorizer(sparse=False))

In [14]:
customer = {
    'contract': 'two_year',
    'tenure': 12,
    'monthlycharges': 19.7
}

In [15]:
def predict_single(customer, dv, model):
    X = dv.transform([customer])
    y_pred = model.predict_proba(X)[:, 1]
    return y_pred[0]

In [16]:
predict_single(customer, dv, model)

0.11549580587832894

**0.115** is the answer

# Question 4 - Probability of churning (Flask) 

In [18]:
import requests

In [19]:
url = 'http://localhost:9696/predict_flask'

In [20]:
customer = {
    "contract": "two_year",
    "tenure": 1,
    "monthlycharges": 10
}

In [21]:
response = requests.post(url, json=customer).json()
print(response)

{'churn': True, 'churn_probability': 0.9988892771007961}


**0.998** is the answer

# Question 5: Digest for the base image (Docker)

sha256:1ee036b365452f8a1da0dbc3bf5e7dd0557cfd33f0e5

# Question 6 - Probability of churning (Docker)

In [1]:
import requests
url = 'http://localhost:9696/predict_flask'
customer = {
    "contract": "two_year",
    "tenure": 12,
    "monthlycharges": 10
}
response = requests.post(url, json=customer).json()
print(response)

{'churn': False, 'churn_probability': 0.32940789808151005}


**0.329** is the answer